mtags/src/main/scala-3/scala/meta/internal/pc/completions/NamedArgCompletions.scala

```diff
@@ -99,60 +112,130 @@ object NamedArgCompletions:
-    // ```
-    // def curry(x; Int)(apple: String, banana: String) = ???
-    // curry(1)(apple = "test", b@@)
-    // ```
-    val (baseParams, baseArgs) =
-      vparamss.zip(argss).lastOption.getOrElse((Nil, Nil))
-
-    val args = ident
-      .map(i => baseArgs.filterNot(_ == i))
-      .getOrElse(baseArgs)
-      .filterNot(isUselessLiteral)
-
-    val isNamed: Set[Name] = args.iterator
-      .zip(baseParams.iterator)
-      // filter out synthesized args and default arg getters
-      .filterNot {
-        case (arg, _) if arg.symbol.denot.is(Flags.Synthetic) => true
-        case (Ident(name), _) => name.is(DefaultGetterName) // default args
-        case (Select(Ident(_), name), _) =>
-          name.is(DefaultGetterName) // default args for apply method
-        case _ => false
-      }
-      .map {
-        case (NamedArg(name, _), _) => name
-        case (_, param) => param.name
-      }
-      .toSet
 
-    val allParams: List[Symbol] =
+    // fallback for when multiple overloaded methods match the supplied args
+    def fallbackFindMatchingMethods() =
+      def maybeNameAndIndexedContext(
+          method: Tree
+      ): Option[(Name, IndexedContext)] =
+        method match
+          case Ident(name) => Some((name, indexedContext))
+          case Select(This(_), name) => Some((name, indexedContext))
+          case Select(from, name) =>
+            val symbol = from.symbol
+            val ownerSymbol =
+              if symbol.is(Method) && symbol.owner.isClass then
+                Some(symbol.owner)
+              else Try(symbol.info.classSymbol).toOption
+            ownerSymbol.map(sym =>
+              (name, IndexedContext(context.localContext(from, sym)))
+            )
+          case Apply(fun, _) => maybeNameAndIndexedContext(fun)
+          case _ => None
+      val matchingMethods =
+        for
+          (name, indxContext) <- maybeNameAndIndexedContext(method)
+          potentialMatches <- indxContext.findSymbol(name)
+        yield potentialMatches.collect {
+          case m
+              if m.is(Flags.Method) &&
+                m.vparamss.length >= argss.length &&
+                m.isAccessibleFrom(apply.symbol.info) &&
+                m.vparamss
+                  .zip(argss)
+                  .reverse
+                  .zipWithIndex
+                  .forall { case (pair, index) =>
+                    FuzzyArgMatcher(m.tparams)
+                      .doMatch(allArgsProvided = index != 0)
+                      .tupled(pair)
+                  } =>
+            m
+        }
+      matchingMethods.getOrElse(Nil)
+    end fallbackFindMatchingMethods
+
+    val matchingMethods: List[Symbols.Symbol] =
+      if method.symbol.paramSymss.nonEmpty
+      then
+        val allArgsAreSupplied =
+          val vparamss = method.symbol.vparamss
+          vparamss.length == argss.length && vparamss
+            .zip(argss)
+            .lastOption
+            .exists { case (baseParams, baseArgs) =>
```

Comment:
Why only check last?

Reply:
So we are checking here is all the arguments in the currently filled arg list are supplied. In all the arg lists before all args had to be supplied for the compiler to even see that this is the method we are calling.
```k.Scala
def m(p: Int, s: String)(k: Int)(l: String)
//                v---v-currently filled arg list
val o = m(1, "aa")(k@@)
```